### PR TITLE
KAFKA-14790: Add more AddPartitionsToTxn tests in KafkaApis and Authorizer tests

### PR DESCRIPTION
**Pull request by jolshan**
Added tests to KafkaApis. We can decide if this is sufficient or we want Authorizer specific tests too.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/server/KafkaApisTest.scala [hunk 1]
```diff
@@ -2113,6 +2113,115 @@ class KafkaApisTest {
...
+
+    val addPartitionsToTxnRequest = if (version < 4) 
+      AddPartitionsToTxnRequest.Builder.forClient(
+        transactionalId,
+        producerId,
+        epoch,
+        Collections.singletonList(tp)).build(version)
+    else
+      AddPartitionsToTxnRequest.Builder.forBroker(
+        new AddPartitionsToTxnTransactionCollection(
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/server/KafkaApisTest.scala]: nit: indentation seems to be off.

File: core/src/test/scala/unit/kafka/server/KafkaApisTest.scala [hunk 2]
```diff
@@ -2113,6 +2113,115 @@ class KafkaApisTest {
...
+    val requestChannelRequest = buildRequest(addPartitionsToTxnRequest)
+
+    val authorizer: Authorizer = mock(classOf[Authorizer])
+    when(authorizer.authorize(any[RequestContext], any[util.List[Action]]))
+      .thenReturn(Seq(AuthorizationResult.DENIED).asJava)
+
+      createKafkaApis(authorizer = Some(authorizer)).handle(
+        requestChannelRequest,
+        RequestLocal.NoCaching
+      )
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/server/KafkaApisTest.scala]: nit: indentation is off.

File: core/src/test/scala/unit/kafka/server/KafkaApisTest.scala [hunk 3]
```diff
@@ -2113,6 +2113,115 @@ class KafkaApisTest {
...
+
+    val response = verifyNoThrottling[AddPartitionsToTxnResponse](requestChannelRequest)
+    val error = if (version < 4) 
+      response.errors().get(AddPartitionsToTxnResponse.V3_AND_BELOW_TXN_ID).get(tp) 
+    else
+      Errors.forCode(response.data().errorCode())
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/server/KafkaApisTest.scala]: nit: You can omit the `()` here.

File: core/src/test/scala/unit/kafka/server/KafkaApisTest.scala [hunk 4]
```diff
@@ -2113,6 +2113,115 @@ class KafkaApisTest {
...
+
+    val response = verifyNoThrottling[AddPartitionsToTxnResponse](requestChannelRequest)
+    
+    def checkErrorForTp(tp: TopicPartition): Unit = {
+      val error = if (version < 4)
+        response.errors().get(AddPartitionsToTxnResponse.V3_AND_BELOW_TXN_ID).get(tp)
+      else
+        response.errors().get(transactionalId).get(tp)
+
+      val expectedError = if (tp == tp0) Errors.OPERATION_NOT_ATTEMPTED else Errors.UNKNOWN_TOPIC_OR_PARTITION
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/server/KafkaApisTest.scala]: nit: Would it make sense to pass the expected error as an argument? The `if (tp == tp0)` is a bit nasty here.

File: core/src/test/scala/unit/kafka/server/KafkaApisTest.scala [hunk 5]
```diff
@@ -2113,6 +2113,115 @@ class KafkaApisTest {
...
+      assertEquals(expectedError, error)
+    }
+    checkErrorForTp(tp1)
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/server/KafkaApisTest.scala]: nit: I would add an empty line before this one.

**Review by dajac — COMMENTED** [review]
@jolshan Thanks for the PR. Left a few nits.

File: core/src/test/scala/unit/kafka/server/KafkaApisTest.scala (hunk 1 above)
Comment (jolshan) [reply core/src/test/scala/unit/kafka/server/KafkaApisTest.scala]: hmm are you referring to where List is?

Comment (dajac) [reply core/src/test/scala/unit/kafka/server/KafkaApisTest.scala]: correct

**Review by dajac — APPROVED** [review]
LGTM, thanks.

**Pull request merged**